pages/docs/concepts/asyncapi-document/adding_messages.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Adding Messages
+weight: 140
+---
+
+[Messages](../message.md) in AsyncAPI represent the data being transmitted between your systems. They always include a payload, which is the data the application is sending or receiving, and they may also include headers, which are used to provide additional context or information about the message exchange.
+
+```mermaid
+graph TD;
+    A[AsyncAPI Message] --> B[Headers]
+    A --> C[Payload]
+    B --> D[Header Content]
+    C --> E[Payload Content]
+
+    style A fill:#47BCEE,stroke:#47BCEE;
+```
+
+Here is an example of a simple message:
+
+```yml
+channels:
+    user/signedup:
+    subscribe:
+        message:
+        $ref: "#/components/messages/userSignUp"
+messages:
+    userSignUp:
+    summary: Action to sign a user up.
+    description: |
+        Multiline description of what this action does.
+        Here you have another line.
+```
+
+This document defines a `user/signedup` channel where a `subscription` to the `userSignUp` message can be made; with this `subscription`.
+
+## Using `oneOf` for Multiple Message Types
```

Comment:
IMHO, from here below, this is too much for a concept document.

Reply:
Nope, we are documenting in detail and the below information is still needed. Some documents are longer, that is not a problem. The goal is to fully document each section, not try to force a "short doc."